browser/resources/settings/email_aliases_page/email_aliases.tsx

```diff
@@ -0,0 +1,248 @@
+          {getLocale('emailAliasesLearnMore')}
+        </SecureLink>
+      </div>
+  </SectionTitle>
+
+const MainEmailDisplay = ({ email, onLogout }:
+  { email: string, onLogout: () => void }) =>
+  <Card>
+    <AccountRow>
+      <Row>
+        <BraveIconCircle name='social-brave-release-favicon-fullheight-color' />
+        <MainEmailTextContainer>
+          <MainEmail>
+            {email === ''
+              ? getLocale('emailAliasesConnectingToBraveAccount')
+              : email}
+          </MainEmail>
+          <MainEmailDescription>
+            {getLocale('emailAliasesBraveAccount')}
+          </MainEmailDescription>
+        </MainEmailTextContainer>
+      </Row>
+      <Button
+        kind='plain-faint'
+        title={getLocale('emailAliasesSignOutTitle')}
+        size='small'
+        onClick={(e) => {
+          onLogout()
+        }}>
+        <Icon slot='icon-before' name="outside" />
+        <span>{getLocale('emailAliasesSignOut')}</span>
+      </Button>
+    </AccountRow>
+  </Card>
+
+const SpacedRow = styled(Row)`
+  gap: ${spacing.m};
+  justify-content: center;
+  align-items: center;
+  font: ${font.default.semibold};
+`
+
+const MainView = ({
+  viewState, mainEmail, onLogout, aliasesState, setViewState,
+  mappingService, onListChange
+}: {
+  viewState: ViewState,
+  mainEmail: string,
+  onLogout: () => void,
+  aliasesState: Alias[],
+  setViewState: (viewState: ViewState) => void,
+  mappingService: MappingService,
+  onListChange: () => void
+}) => (viewState.mode === 'Startup'
+        ? <SpacedRow>
+            <LoadingIcon />
+            <div>{getLocale('emailAliasesConnectingToBraveAccount')}</div>
+          </SpacedRow>
+        : <span>
+            <MainEmailDisplay onLogout={onLogout} email={mainEmail} />
+            <AliasList aliases={aliasesState}
+              onViewChange={setViewState}
+              mappingService={mappingService}
+              onListChange={onListChange} />
+          </span>)
+
+export const ManagePage = ({ mappingService }:
+  {
+    mappingService: MappingService
+  }) => {
+  const [viewState, setViewState] = React.useState<ViewState>(
+    { mode: 'Startup' })
+  const [mainEmail, setMainEmail] = React.useState<string>('')
+  const [aliasesState, setAliasesState] = React.useState<Alias[]>([]);
+  const onEmailChange = async () => {
+    const email = await mappingService.getAccountEmail()
+    setMainEmail(email ?? '')
+    setViewState({ mode: email ? 'Main' : 'SignUp' })
+  }
+  const onListChange = async () => {
+    const aliases = await mappingService.getAliases()
+    setAliasesState(aliases)
+  }
+  const onMainEmailSubmitted = async (email: string) => {
+    setMainEmail(email)
+    await mappingService.requestAccount(email)
+    setViewState({ mode: 'AwaitingAuthorization' })
+    const accountReady = await mappingService.onAccountReady()
+    if (accountReady) {
+      setViewState({ mode: 'Main' })
+      await onListChange()
```

Comment:
similarly, with a listener we wouldn't need to worry about calling this all over the place

Reply:
additionally, this looks like `onReturnToMain`

Reply:
Please see my comment above.

Reply:
moved to an observer model